lib/matplotlib/dates.py

```diff
@@ -1283,7 +1283,6 @@ def __init__(self, tz=None, minticks=5, maxticks=None,
             at 6 hour intervals.
         """
         DateLocator.__init__(self, tz)
-        self._locator = YearLocator(tz=tz)
```

Comment:
`self._locator` is assigned in `autoscale`, which is fine since it'll never be read, but I guess you can remove the assignment there too.

Reply:
fair point.